packages/web/src/components/organisms/TriggersFormItems/ResourceTriggerSelect/ResourceTriggerSelect.tsx

```diff
@@ -24,38 +24,46 @@ import {StyledResourceOptionWrapper} from './ResourceTriggerSelect.styled';
 const {Option, OptGroup} = Select;
 
 interface ResourceTriggerSelectProps {
+  $allowTests: boolean;
+  $allowTestSuites: boolean;
   disabled?: boolean;
   value?: string;
   onChange?: (value: string) => void;
 }
 
-const ResourceTriggerSelect: FC<ResourceTriggerSelectProps> = ({...props}) => {
+const ResourceTriggerSelect: FC<ResourceTriggerSelectProps> = props => {
+  const {$allowTestSuites, $allowTests, disabled, value, onChange, ...rest} = props;
+
   const isClusterAvailable = useSystemAccess(SystemAccess.agent);
 
   const {executors = []} = useExecutorsPick('executors');
 
-  const {data: tests = []} = useGetAllTestsQuery(null, {skip: !isClusterAvailable});
+  const {data: tests = []} = useGetAllTestsQuery(null, {skip: !isClusterAvailable || !$allowTests});
   const {data: testSuites = []} = useGetAllTestSuitesQuery(null, {
```

Comment:
One bug that was already before - `tests = []` and `testSuites = []` should be avoided, as both `tests` and `testSuites` are used later for `useMemo` dependencies. Having `tests = []` here, will always result in recomputing, as `[]` will always be a new array.